osquery/core/watcher.cpp

```diff
@@ -287,17 +287,21 @@ void WatcherRunner::watchExtensions() {
 
       // The extension manager also watches for extension-related failures.
       // The watchdog is more general, but may find failed extensions first.
-      createExtension(extension.first);
-      extension_restarts_[extension.first] += 1;
+      if (extension_backoff_[extension.first] == 0) {
+        createExtension(extension.first);
+        ++extension_restarts_[extension.first];
+        extension_backoff_[extension.first] =
```

Comment:
This is going to be weird when debugging, basically it will possibly print `... back off set to ...` after the extension start. Maybe set the back outside the if as soon as you know the extension failed to start. Here log that you're re-trying to create it.